testsuite/verona-mlir/struct.verona

```diff
@@ -50,7 +57,7 @@ foo() : I32
   boop.d = PI;
 
   // Test reading / writing to fields
-  foo.a = 12;
+  foo.a = I64::trunc(12);
```

Comment:
I would expect type inference to start with a set of possible types for the integer literal and then to define the final type as `I32` based on the assignment.  If there is a bug preventing that in the front end, please can you leave a FIXME in the test so that it can be cleaned up later?

Reply:
It will be, but for now, AST nodes don't have types, so I have to define some type, and I chose 64-bit types for both integer and floating point literals. There is a FIXME on the code for that already.